### ISSUE TITLE
"from" problem and its solution

### ISSUE THREAD

**Issue by mnakalay** [aklump/breakpointX#1]
Hello,

This is pretty neat and works nicely except for the "from" parameter. it always has the same value as the "to" parameter so as a result, the corresponding class never gets removed from the element it is added to. And it's value is always wrong.

The reason is because around line 253 you have:

```
           // Update for next round.
            self.last = {
              "width"    : self.value(currentAlias),
              "alias"    : currentAlias,
              "direction": direction
            };
            self.current = currentAlias;

            // Fire off all callbacks.
            for (var i in callbacks) {
                var from = {
                    minWidth: self.breakpoints[self.last.alias][0],
                    maxWidth: self.breakpoints[self.last.alias][1],
                    name: self.last.alias
                };
                var to = {
                    minWidth: self.breakpoints[currentAlias][0],
                    maxWidth: self.breakpoints[currentAlias][1],
                    name: currentAlias
                };
                for (var j in callbacks[i][breakpoint]) {
                    callbacks[i][breakpoint][j].call(self, from, to, direction);
                }
            }
```

So by the time you get to the for loop, self.last has already been given the same value as self.current.

To make it work you should assign the current value to self.last AFTER the for loop like so:

```
            self.current = currentAlias;

            // Fire off all callbacks.
            for (var i in callbacks) {
                var from = {
                    minWidth: self.breakpoints[self.last.alias][0],
                    maxWidth: self.breakpoints[self.last.alias][1],
                    name: self.last.alias
                };
                var to = {
                    minWidth: self.breakpoints[currentAlias][0],
                    maxWidth: self.breakpoints[currentAlias][1],
                    name: currentAlias
                };
                for (var j in callbacks[i][breakpoint]) {
                    callbacks[i][breakpoint][j].call(self, from, to, direction);
                }
            }
            // Update for next round.
            self.last = {
              "width"    : self.value(currentAlias),
              "alias"    : currentAlias,
              "direction": direction
            };
```

Hope it helps :)

**Comment by aklump** [comment]
@mnakalay Thank you for finding this and letting me know.  I know there is another issue that this brings up, so I'll have to refactor that as well.  I'll post an update as soon as possible.

**Comment by mnakalay** [comment]
Really? I didn't notice anything. I tried it with different elements and all and it seems to be working pretty well

**Comment by aklump** [comment]
I believe it was a change to make defaultTextCallback work, but I'll have to look it up.

**Comment by aklump** [comment]
I have fixed this in 0.4.2; thanks again for pointing this out @mnakalay
